uaclient/snap.py

```diff
@@ -133,3 +142,94 @@ def get_installed_snaps() -> List[SnapPackage]:
         snaps.append(snap)
 
     return snaps
+
+
+def install_snapd():
+    event.info(messages.APT_UPDATING_LISTS)
+    try:
+        apt.run_apt_update_command()
+    except exceptions.UserFacingError as e:
+        logging.debug(
+            "Trying to install snapd." " Ignoring apt-get update failure: %s",
+            str(e),
+        )
+    try:
+        system.subp(
+            ["apt-get", "install", "--assume-yes", "snapd"],
+            retry_sleeps=apt.APT_RETRIES,
+        )
+    except exceptions.ProcessExecutionError:
+        raise exceptions.CannotInstallSnapdError()
+
+
+def run_snapd_wait_cmd():
+    try:
+        system.subp([SNAP_CMD, "wait", "system", "seed.loaded"], capture=True)
+    except exceptions.ProcessExecutionError as e:
+        if re.search(r"unknown command .*wait", str(e).lower()):
+            logging.warning(messages.SNAPD_DOES_NOT_HAVE_WAIT_CMD)
+        else:
+            raise
+
+
+def install_snap(snap: str, channel: Optional[str] = None):
+    cmd = [SNAP_CMD, "install", snap]
+
+    if channel:
+        cmd += ["--channel", channel]
+
+    system.subp(
+        cmd,
+        capture=True,
+        retry_sleeps=SNAP_INSTALL_RETRIES,
+    )
+
+
+def get_snap_info(snap: str) -> SnapInfo:
+    snap_sock = socket.socket(socket.AF_UNIX, socket.SOCK_STREAM)
+    snap_sock.connect(SNAPD_SOCKET_PATH)
+
+    conn = http.client.HTTPConnection("localhost")
```

Comment:
This usage of the actual API is great! I think should replace the usage of the `get_snap_package_info_tracking` function above

Reply:
Done